File: ios/SuperheroApp/Characters/CharactersContract.swift
//
//  CharactersContract.swift
//  SuperheroApp
//
//  Created by Balazs Varga on 2018. 02. 12..
//  Copyright © 2018. W.UP. All rights reserved.
//

import Common

protocol CharactersMvpPresenter: MvpPresenter {

    func takeView(view: CharactersMvpView)

    func characters() -> [CommonCharacter]

    func charactersCount() -> Int

    func loadCharacters()

    func loadMoreCharacters()
}

protocol CharactersMvpView: MvpView {

    func showLoadingIndicator()

    func hideLoadingIndicator()

    func showMoreLoadingIndicator()

    func hideMoreLoadingIndicator()

    func refreshCharacters()

    func showLoadingCharactersError(message: String)

    func showNoCharacters()
}

File: common/src/main/kotlin/digital/wup/superheroapp/characters/CharactersContract.kt
package digital.wup.superheroapp.characters

import digital.wup.superheroapp.characters.domain.usecase.model.Character
import digital.wup.superheroapp.mvp.MvpPresenter
import digital.wup.superheroapp.mvp.MvpView

interface CharactersMvpPresenter: MvpPresenter {

    fun takeView(view: CharactersMvpView)

    fun characters(): List<Character>

    fun charactersCount(): Int

    fun loadCharacters()

    fun loadMoreCharacters()
}

interface CharactersMvpView: MvpView {

    fun showLoadingIndicator()

    fun hideLoadingIndicator()

    fun showMoreLoadingIndicator()

    fun hideMoreLoadingIndicator()

    fun refreshCharacters()

    fun showLoadingCharactersError(message: String)

    fun showNoCharacters()
}